pkg/sfu/downtrack.go

```diff
@@ -550,7 +558,13 @@ func (d *DownTrack) keyFrameRequester(generation uint32, layer int32) {
 	}
 	ticker := time.NewTicker(time.Duration(interval) * time.Millisecond)
 	defer ticker.Stop()
+
 	for {
+		locked, _ := d.forwarder.CheckSync()
+		if locked {
+			return
+		}
+
```

Comment:
This is the change to check for layer lock and quick the key frame requester. Previously, we were relying on a return from video layer selector that we have switched to proper layer and stopping the key frame requester. Was more complex. Simplifying by checking inline here.